File: src/br/edu/ifrn/obi2012/Campeonato.java
package br.edu.ifrn.obi2012;

// http://olimpiada.ic.unicamp.br/pdf/provas/ProvaOBI2012_prog_f1nj.pdf

import java.util.Scanner;

public class Campeonato {
    public static void main(String[] args){
        
        Scanner input = new Scanner(System.in);
        
        String[] entradas = input.nextLine().split(" ");

        int cVitorias = Integer.valueOf(entradas[0]);
        int cEmpates = Integer.valueOf(entradas[1]);
        int cGols = Integer.valueOf(entradas[2]);
        int fVitorias = Integer.valueOf(entradas[3]);
        int fEmpates = Integer.valueOf(entradas[4]);
        int fGols = Integer.valueOf(entradas[5]);
        
        int cPontos = cVitorias * 3 + cEmpates;
        int fPontos = fVitorias * 3 + fEmpates;
        
        String saida = cPontos == fPontos ? (cGols == fGols ? "=" : (cGols > fGols ? "C" : "F")) : (cPontos > fPontos ? "C" : "F");
        
        System.out.println(saida);
        
    }
}

File: obi/2012/campeonato.cpp
// http://olimpiada.ic.unicamp.br/pdf/provas/ProvaOBI2012_prog_f1nj.pdf

#include <iostream>

using namespace std;

int main() {
    
    // dados do time C
    int cPontos;
    int cVitorias, cEmpates, cGols;
    
    // dados do time F
    int fPontos;
    int fVitorias, fEmpates, fGols;

    
    // inputs
    cin >> cVitorias;
    cin >> cEmpates;
    cin >> cGols;
        
    cin >> fVitorias;
    cin >> fEmpates;
    cin >> fGols;

    
    // cálculo de pontos
    cPontos = cVitorias * 3 + cEmpates;
    fPontos = fVitorias * 3 + fEmpates;

    
    // output de vencedor.
    // O vencedor é o que tiver maior pontos.
    // Se os pontos forem iguais, a decisão é de quem tiver maior gols.
    cout << (
        
        (cPontos == fPontos) ? // if (empate de pontos)
            
            // then decisão por pontos inviável: checar gols
            (cGols == fGols ? // o empate também é de gols?
                "="
             : // gols diferentes, então maior ngols vence
                (cGols > fGols ? "C" : "F"))
        
        : // if not (empate de pontos)

            // then decisão por pontos é possível: o maior vence
            (cPontos > fPontos ? "C" : "F")
        
    ) << endl; // fim do cout

    
    return 0;

}
